### ISSUE TITLE
Notification for deleted wiki page version shows wrong username

### ISSUE THREAD

**Issue by raimue** [trac-hacks/trac-wikinotification#11]
When deleting a version of a wiki page, the plugin will send a notification like the following:

    Version "42" of page "Example" was deleted by foo

However, the "foo" user listed here is the user who made the original change and not the user who deleted the page as the wording suggests. To be consistent with the notification for completely deleting a wiki page, it should name the user who clicked the delete button.

The problem appears to originate in the following code snippet. I guess `req.authname` should be used as in `wiki_page_deleted()` a few lines above?
https://github.com/trac-hacks/trac-wikinotification/blob/e9377ae073079282a1ff2986068bb311d7cd439e/WikiNotification/listener.py#L62-L66

This issue was originally reported at the [MacPorts Trac](https://trac.macports.org/ticket/40190).

**Comment by rjollos** [comment]
The `Request` object isn't passed to the change listener, so it's not possible to know the user that deleted the page. A change would need to be made to Trac. Discussed in [#8859](https://trac.edgewall.org/ticket/8859#comment:5) and [#6744](https://trac.edgewall.org/ticket/6744).

**Comment by rjollos** [comment]
Created [#12928](https://trac.edgewall.org/ticket/12928) with a possible solution.

**Comment by raimue** [comment]
I agree that the best solution would be that Trac provides this information. Thank you for passing this on.

However, what you linked is discussing `wiki_page_deleted()`, but the [current implementation] of TracWikiNotification (https://github.com/trac-hacks/trac-wikinotification/blob/e9377ae073079282a1ff2986068bb311d7cd439e/WikiNotification/listener.py#L55-L60) is already able to obtain a `Request` object to get the username. The way this is done looks like a crude hack, though.

As the hack mentioned above is already in use for the other notifications, I think it could also be used for `wiki_page_version_deleted()` as a fix for consistency. Digging a bit deeper, I found out that the original intention until f81d125fe2f465a6ab86943248b9ba593a93e677 was probably to use the same approach, but the variables were overwritten with the page history in the next line.

Looking at the page history is definitely wrong, as it no longer contains the deleted page version at this point. Maybe in the past this worked because the notification callback was run before the page version was removed from the history.
